backend/app/utils/network.py

```
@@ -0,0 +1,115 @@
+    route_ip = network_util_default_route_ip()
+    stats = psutil.net_if_stats()
+    candidates: List[InterfaceCandidate] = []
+
+    for interface, addrs in psutil.net_if_addrs().items():
+        for addr in addrs:
+            if addr.family != socket.AF_INET:
+                continue
+            ip = addr.address
+            # Loopback cannot be reached from another device, and a 169.254
+            # address means DHCP never answered — a symptom, not an address.
+            if ip.startswith("127.") or ip.startswith("169.254."):
```

Ruff (0.16.1)

[warning] Call `startswith` once with a `tuple`

Merge into a single `startswith` call

(PIE810)